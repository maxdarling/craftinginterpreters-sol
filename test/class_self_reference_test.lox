// Test to check if class name is accessible within methods

print "=== Testing Class Name Access Within Methods ===";

class SelfReference {
  init() {
    this.name = "instance";
  }

  getName() {
    return this.name;
  }

  getClassName() {
    // This should work - accessing the class name from within a method
    return SelfReference; // Should return the class object
  }

  createAnother() {
    // This should also work - creating another instance from within a method
    return SelfReference();
  }
}

var instance = SelfReference();
print instance.getName(); // Should print "instance"
print instance.getClassName(); // Should print "SelfReference"
var another = instance.createAnother();
print another; // Should print "SelfReference instance"

print "Test completed";
