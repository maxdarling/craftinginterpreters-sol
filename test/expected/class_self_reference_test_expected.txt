=== Testing Class Name Access Within Methods ===
instance
SelfReference
SelfReference instance
Test completed
